pkg/cert-generator/v1beta1/generate/generate.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2021 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generate
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"github.com/kubeflow/katib/pkg/cert-generator/v1beta1/consts"
+	"github.com/spf13/cobra"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"math/big"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"time"
+)
+
+// generateOptions contains values for all certificates.
+type generateOptions struct {
+	namespace         string
+	fullServiceDomain string
+}
+
+// NewGenerateCmd sets up `generate` subcommand.
+func NewGenerateCmd(kubeClient client.Client) *cobra.Command {
+	o := &generateOptions{}
+	cmd := &cobra.Command{
+		Use:          "generate",
+		Short:        "generate server cert for webhook",
+		Long:         "generate server cert for webhook",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.run(context.TODO(), kubeClient); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
```

Comment:
Sorry that I haven't asked about it before.
Since we have only 1 entrypoint command `generate`, do we need `cobra` command in this script ?
Are we planing to extend this script in the future with additional commands (e.g. `update`) ?

If not, we might simplify it to use just command line arguments (similar to [`katib-controller`](https://github.com/kubeflow/katib/blob/master/cmd/katib-controller/v1beta1/main.go#L52-L65)).

What do you think @tenzen-y @gaocegege ?

Reply:
I'm not planning it.
I implemented using cobra because I thought it would give me better visibility.

Reply:
Sounds good.
What do you think about it @johnugeorge @gaocegege ?

Reply:
It looks ok to me

Reply:
LGTM